packages/office-ui-fabric-react/src/components/pickers/BasePicker.tsx

```diff
@@ -106,6 +106,13 @@ export class BasePicker<T, P extends IBasePickerProps<T>> extends React.Componen
   private _id: string;
   private _async: Async;
 
+  public static getDerivedStateFromProps(newProps: IBasePickerProps<any>) {
```

Comment:
This unconditionally overwrites `state.items` with `props.items` **on every render**, not just on prop updates--which means setting `state.items` in `_updateSelectedItems` will no longer do anything. So you'll probably need a different approach here.

Reply:
For this reason, in some cases moving logic to `componentDidUpdate` under a `this.props !== prevProps` check provides a closer approximation of old `componentWillReceiveProps` behavior. This means technically you'd have slightly-mismatched data for one render, but IIRC the render with mismatched data will never actually be shown because React flushes any additional renders caused by `componentDidUpdate` before actually updating the DOM. So it's a bit of a perf hit, but likely not a practical problem unless it causes timing issues with side effects in other lifecycle methods.

Reply:
This is correct behavior I believe. If selected items are provided, then this is a controlled picker and the incoming `selectedItems` should be the _only_ source of truth.